### PR TITLE
feat(query): support unset session setting

### LINKED ISSUE #15498: feat: unset session <session>
**Summary**

Databend now supports the `unset <settings>` command, which is designed to remove global settings from the metasrv. We plan to expand this functionality with additional unset commands:

```sql
unset [session] <setting>; -- Only unset a setting at the session level.
```

For example:
```
-- load_file_metadata_expire_hours default is 248

set global load_file_metadata_expire_hours=12;
-- show settings like '%load_file_metadata_expire_hours%';
-- 12 global-level

set load_file_metadata_expire_hours=13;
-- show settings like '%load_file_metadata_expire_hours%';
-- 13 session-level

unset session load_file_metadata_expire_hours;
-- show settings like '%load_file_metadata_expire_hours%';
-- 12 global-level
```

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Databend now supports the `unset <settings>` command, which is designed to remove global settings from the metasrv. We plan to expand this functionality with additional unset commands:

```sql
unset [session] <setting>; -- Only unset a setting at the session level.
```

For example:

```sql
-- load_file_metadata_expire_hours default is 248

set global load_file_metadata_expire_hours=12;
-- show settings like '%load_file_metadata_expire_hours%';
-- 12 global-level

set load_file_metadata_expire_hours=13;
-- show settings like '%load_file_metadata_expire_hours%';
-- 13 session-level

unset session load_file_metadata_expire_hours;
-- show settings like '%load_file_metadata_expire_hours%';
-- 12 global-level
```

- Fixes #15498

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15510)

### CODE REVIEW

**Comment by TCeason** [comment]
Add new token SESSION. cc @andylokandy  Please review this part.

Modify unset interpreter, cc @zhang2014 Please review this part.

**Comment by TCeason** [comment]
After https://github.com/datafuselabs/databend/pull/15511 merged , re-run failed ci

File: tests/sqllogictests/suites/base/05_ddl/05_0026_ddl_unset_settings.test [hunk 1]
```diff
@@ -58,3 +50,34 @@ show settings like 'DATA_RETENTION_TIME_IN_DAYS';
...
+query TTTTTTT
+show settings like 'load_file_metadata_expire_hours';
+----
+load_file_metadata_expire_hours 12 168 [0, 18446744073709551615] GLOBAL Sets the hours that the metadata of files you load data from with COPY INTO will expire in. UInt64
+
+statement ok
+unset load_file_metadata_expire_hours
+
+query TTTTTTT
+show settings like 'load_file_metadata_expire_hours';
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/base/05_ddl/05_0026_ddl_unset_settings.test:81]: `load_file_metadata_expire_hours` default value is changed:  https://github.com/datafuselabs/databend/pull/15514

Comment (TCeason) [reply tests/sqllogictests/suites/base/05_ddl/05_0026_ddl_unset_settings.test:81]: Fixed

**Review by zhang2014 — APPROVED** [review]

**Comment by TCeason** [comment]
Cc @soyeric128 doc needed

**Pull request merged**